src/state.rs

```diff
@@ -1,52 +1,132 @@
-use super::gui::GUI;
+use super::gui::Gui;
+// Move these to preludes later
+use super::memory::MemoryManagers;
+use memory::game_engine::il2cpp::{Image, Module};
 use memory::process::Process;
 use memory::process_list::ProcessList;
+use std::time::Instant;
+use crate::gui::helpers::GuiHelper;
+use egui_dock::DockState;
+
+pub struct StateDebug {
+    pub last_update: Instant,
+    pub last_memory_update: Instant,
+}
+
+pub struct StateGui {
+    pub helpers: Vec<Box<dyn GuiHelper>>,
+    pub dock_state: DockState<String>
+}
 
 pub struct State {
-    pub process: Option<Process>,
-    pub module: u64,
+    pub context: StateContext,
     pub process_list: ProcessList,
+    pub memory_managers: MemoryManagers,
+    pub debug: StateDebug,
+    pub gui: StateGui
+}
+
+pub struct StateContext {
+    pub process: Option<Process>,
+    pub module: Option<Module>,
+    pub image: Option<Image>,
 }
 
 impl State {
-    pub fn new(cc: &eframe::CreationContext<'_>) -> Self {
-        // Customize egui here with cc.egui_ctx.set_fonts and cc.egui_ctx.set_visuals.
-        // Restore app state using cc.storage (requires the "persistence" feature).
-        // Use the cc.gl (a glow::Context) to create graphics shaders and buffers that you can use
-        // for e.g. egui::PaintCallback.
+    pub fn new(_cc: &eframe::CreationContext<'_>) -> Self {
+        // Register any GUI helpers here
+        let gui_helpers: Vec<Box<dyn GuiHelper>> = vec![
+            Box::new(super::gui::helpers::title::TitleHelper::default()),
+            Box::new(super::gui::helpers::main::MainHelper::default()),
+        ];
+
+        let mut tree_names: Vec<String> = vec![]; 
 
+        for h in gui_helpers.iter() {
+            tree_names.push(h.name());
+        }
+            
         Self {
```

Comment:
So if I understand correctly, the `new()` function here will set up the internal state structure, but not initialize it (so at this point, we have a gui and an app running, but the game isn't necessarily running and we are not connected to the memory process.